app/src/org/commcare/connect/ConnectManager.java

```diff
@@ -315,23 +322,30 @@ public static ConnectJobRecord getActiveJob() {
 
     public static void unlockConnect(CommCareActivity<?> parent, ConnectActivityCompleteListener listener) {
         if(manager.connectStatus == ConnectIdStatus.LoggedIn) {
-            ConnectIdWorkflows.unlockConnect(parent, success -> {
-                if(success) {
-                    completeSignin();
-                }
-                listener.connectActivityComplete(success);
-            });
+            Intent intent=new Intent(parent,ConnectIdActivity.class);
+            Bundle bundle=new Bundle();
+            intent.putExtra("TASK",ConnectConstants.UNLOCK_CONNECT);
```

Comment:
Let's create a common launchConnectId(parent, ConnectConstants task) function so we don't have to repeat the details of the intent (like "TASK")